### PR TITLE
check intersectionObserver exsits before destroy it

### PR DESCRIPTION
**Pull request by zhouyx**

### CODE REVIEW

File: extensions/amp-ad/0.1/amp-ad-api-handler.js [hunk 1]
```diff
@@ -145,8 +145,10 @@ export class AmpAdApiHandler {
     }
     // IntersectionObserver's listeners were cleaned up by
     // setInViewport(false) before #unlayoutCallback
-    this.intersectionObserver_.destroy();
-    this.intersectionObserver_ = null;
+    if (this.intersectionObserver_) {
```
Comment (jridgewell) [inline extensions/amp-ad/0.1/amp-ad-api-handler.js]: No need to check here, it's guaranteed if `iframe_` is present.

**Comment by jridgewell** [comment]
Looks like `<amp-iframe>` is the only one that needs this check. LGTM afterwards.

**Comment by jridgewell** [comment]
Fixes #4372.

**Comment by zhouyx** [comment]
you're right. amp-ad will create intersectionObserver anyway. removed check for ad

**Pull request merged**